changelog.d/13569.misc

```diff
@@ -0,0 +1 @@
+Remove cache on `_get_joined_users_from_context` calls that are already cached elsewhere. Contributed by Nick @ Beeper (@fizzadar).
```

Comment:
I think we should make this more explicit, somehow, since it means that any local configuration of the cache size is redundant, so admins may wish to update their config.

Like, rephrase it as "Remove redundant `_get_joined_users_from_context` cache", and make it a `.removal` rather than a `.misc`?

Reply:
https://github.com/matrix-org/synapse/pull/13569/commits/d5a5f7f49547b8fdb5a44bc7e22b19cb4b505210